### ISSUE TITLE
after update on 6.12.3, xy.Should().BeEquivalentTo cannot be used any more, has exception

### ISSUE THREAD

**Issue by D-Maier** [fluentassertions/fluentassertions#2852]
### Description

i use the method to test the equality of properties between two objects of the same class. Under version 6.12.2, it works flawlessly, but in version 6.12.3, the following exception occurs:
```
CS0176 Member 'object.Equals(object?, object?)' cannot be accessed with an instance reference; qualify it with a type name instead.
```

### Reproduction Steps

```csharp
_address.Should().BeEquivalentTo(
    compareAddress,
    o => o.Excluding(x => x.City).Excluding(x => x.CityReference).Excluding(x => x.ID)
    );
```

### Expected behavior

Should return true or false.

### Actual behavior

```
Exception CS0176 Member 'object.Equals(object?, object?)' cannot be accessed with an instance reference; qualify it with a type name instead. 
```

### Are you willing to help with a pull-request?

No

**Comment by jnyrup** [comment]
Please provide a complete example that we can run.

**Comment by D-Maier** [comment]
> Please provide a complete example that we can run.

Sorry, the address is actually quite a complex object here, integrated into a large project, probably difficult to submit here in compiled form.
        var compareAddress = Address.CreateNewInstance(true, "Addition", "Street", _city, null, null);
        var address2 = Address.CreateNewInstance(true, "Addition", "Street", _city, null, null);
        compareAddress.Should().BeEquivalentTo( address2);


Interestingly, this call works without any problems:

            "hello".Should().BeEquivalentTo(
               "hello"
                );

**Comment by jnyrup** [comment]
That stacktrace looks like a duplicate of #2832

**Comment by D-Maier** [comment]
> That stacktrace looks like a duplicate of #2832

Okay, I understand. And is it already clear if and when the problem will be fixed or should I replace the method with other calls?

**Comment by jnyrup** [comment]
We intend to publish a new version with the fix #2832 soon-ish - no ETA though.

If the previous version worked for you, then I recommend you rollback to that for now.

**Comment by IT-VBFK** [comment]
You must be ahead of time since 6.12.3 is not released ;)

**Comment by D-Maier** [comment]
> You must be ahead of time since 6.12.3 is not released ;)

my mistake, 6.12.1 (!) works fine, exception starts with 6.12.2

**Comment by dennisdoomen** [comment]
Is fixed by #2836 and will be released soon

**Comment by D-Maier** [comment]
> Is fixed by #2836 and will be released soon
Great, version 7 seems to have fixed the problem, thanks for your great work!
